mne/utils.py

```diff
@@ -3172,3 +3172,354 @@ def _check_if_nan(data, msg=" to be plotted"):
+        reason: str
+            entry in `drop_log` for unselected epochs
+        copy: bool
+            return a copy of the current object
+        drop_event_id: bool
+            remove non-existing event-ids after selection
+        select_data: bool
+            apply selection to data
+            (use `select_data=False` if subclasses do not have a
+             valid `_data` field)
+        return_indices: bool
+            return the indices of selected epochs from the original object)
+            in addition to the new `Epochs` objects
+        Returns
+        -------
+        `Epochs` or tuple(Epochs, np.ndarray) if `return_indices` is True
+
+        object with subset of epochs (and optionally array with kept
+        epoch indices)
+        """
+        data = self._data
+        del self._data
+        inst = self.copy() if copy else self
+        self._data, inst._data = data, data
+        del self
+
+        if isinstance(item, str):
+            item = [item]
+
+        # Convert string to indices
+        if isinstance(item, (list, tuple)) and len(item) > 0 and \
+                isinstance(item[0], str):
+            select = inst._keys_to_idx(item)
+        elif isinstance(item, slice):
+            select = item
+        else:
+            select = np.atleast_1d(item)
+            if len(select) == 0:
+                select = np.array([], int)
+        has_selection = hasattr(inst, 'selection')
+        if has_selection:
+            key_selection = inst.selection[select]
+            if reason is not None:
+                for k in np.setdiff1d(inst.selection, key_selection):
+                    inst.drop_log[k] = [reason]
+            inst.selection = key_selection
+
+        inst.events = np.atleast_2d(inst.events[select])
+        if inst.metadata is not None:
+            pd = _check_pandas_installed(strict=False)
+            if pd is not False:
+                metadata = inst.metadata.iloc[select]
+            else:
+                metadata = np.array(inst.metadata, 'object')[select].tolist()
+
+            # will reset the index for us
+            GetEpochsMixin.metadata.fset(inst, metadata, verbose=False)
+            if has_selection:
+                inst.metadata.index = inst.selection
+        if inst.preload and select_data:
+            # ensure that each Epochs instance owns its own data so we can
+            # resize later if necessary
+            inst._data = np.require(inst._data[select], requirements=['O'])
+        if drop_event_id:
+            # update event id to reflect new content of inst
+            inst.event_id = dict((k, v) for k, v in inst.event_id.items()
+                                   if v in inst.events[:, 2])
+        if return_indices:
+            return inst, select
+        else:
+            return inst
+
+    def _keys_to_idx(self, keys):
+        """Find entries in event dict."""
+        keys = [keys] if not isinstance(keys, (list, tuple)) else keys
```

Comment:
`keys = keys if isinstance(keys, (list, tuple)) else [keys]` right?

Reply:
These are ultimately the same, right? Happy to change it, but I don't see the advantage of one over the other.

Reply:
Not having the not is simpler.